lib/rssbridge.php

```diff
@@ -12,78 +12,68 @@
 
 /** Path to the default whitelist file */
-define('WHITELIST_DEFAULT', PATH_ROOT . 'whitelist.default.txt');
+const WHITELIST_DEFAULT = __DIR__ . '/../whitelist.default.txt';
 
 /** Path to the configuration file */
-define('FILE_CONFIG', PATH_ROOT . 'config.ini.php');
+const FILE_CONFIG = __DIR__ . '/../config.ini.php';
 
 /** Path to the default configuration file */
-define('FILE_CONFIG_DEFAULT', PATH_ROOT . 'config.default.ini.php');
+const FILE_CONFIG_DEFAULT = __DIR__ . '/../config.default.ini.php';
 
 /** URL to the RSS-Bridge repository */
-define('REPOSITORY', 'https://github.com/RSS-Bridge/rss-bridge/');
-
-// Interfaces
-require_once PATH_LIB . 'ActionInterface.php';
-require_once PATH_LIB . 'BridgeInterface.php';
-require_once PATH_LIB . 'CacheInterface.php';
-require_once PATH_LIB . 'FormatInterface.php';
-
-// Classes
-require_once PATH_LIB . 'FactoryAbstract.php';
-require_once PATH_LIB . 'FeedItem.php';
-require_once PATH_LIB . 'Debug.php';
-require_once PATH_LIB . 'Exceptions.php';
-require_once PATH_LIB . 'FormatFactory.php';
-require_once PATH_LIB . 'FormatAbstract.php';
-require_once PATH_LIB . 'BridgeFactory.php';
-require_once PATH_LIB . 'BridgeAbstract.php';
-require_once PATH_LIB . 'FeedExpander.php';
-require_once PATH_LIB . 'CacheFactory.php';
-require_once PATH_LIB . 'Authentication.php';
-require_once PATH_LIB . 'Configuration.php';
-require_once PATH_LIB . 'BridgeCard.php';
-require_once PATH_LIB . 'BridgeList.php';
-require_once PATH_LIB . 'ParameterValidator.php';
-require_once PATH_LIB . 'ActionFactory.php';
-require_once PATH_LIB . 'ActionAbstract.php';
-require_once PATH_LIB . 'XPathAbstract.php';
-
-// Functions
-require_once PATH_LIB . 'html.php';
-require_once PATH_LIB . 'error.php';
-require_once PATH_LIB . 'contents.php';
-require_once PATH_LIB . 'php8backports.php';
-
-// Vendor
-define('MAX_FILE_SIZE', 10000000); /* Allow larger files for simple_html_dom */
-require_once PATH_LIB_VENDOR . 'parsedown/Parsedown.php';
-require_once PATH_LIB_VENDOR . 'php-urljoin/src/urljoin.php';
-require_once PATH_LIB_VENDOR . 'simplehtmldom/simple_html_dom.php';
+const REPOSITORY = 'https://github.com/RSS-Bridge/rss-bridge/';
+
+// Allow larger files for simple_html_dom
+const MAX_FILE_SIZE = 10000000;
+
+$files = [
+	__DIR__ . '/Exceptions.php',
+	__DIR__ . '/html.php',
+	__DIR__ . '/error.php',
+	__DIR__ . '/contents.php',
+	__DIR__ . '/php8backports.php',
+	__DIR__ . '/../vendor/parsedown/Parsedown.php',
+	__DIR__ . '/../vendor/php-urljoin/src/urljoin.php',
+	__DIR__ . '/../vendor/simplehtmldom/simple_html_dom.php',
+];
+
+foreach ($files as $file) {
+	require $file;
+}
+
+spl_autoload_register(function ($class) {
+	$folders = [
+		__DIR__ . '/../actions/',
+		__DIR__ . '/../bridges/',
+		__DIR__ . '/../caches/',
+		__DIR__ . '/../formats/',
+		__DIR__ . '/../lib/',
+	];
+	foreach ($folders as $folder) {
+		$file = $folder . $class . '.php';
+		if (is_file($file)) {
+			require $file;
```

Comment:
Is there a reason to use `require` instead of `require_once` here?

Reply:
We get exception if it's trying to include a class definition a second time (which is a bug).

Reply:
Yeah, sounds like bug to me too.

That said, it seems that the autoloader is only called once for each class anyway, which is why `require_once` makes no difference (actually, it is technically slower than `require`).

This is also explained here: https://www.php.net/manual/en/language.oop5.autoload.php#89448

> You should not have to use require_once inside the autoloader, as if the class is not found it wouldn't be trying to look for it by using the autoloader.

Reply:
It's probably a bug if this happens so it's nice to get exception for it. I always use `require`.